### PR TITLE
Add new endpoint to get CVEs by agent.

### LINKED ISSUE #7520: Implement the necessary changes in the framework to make a user capable of query the CVEs that affect each agent through the RESTful API
### Description

As part of the epic #7479, we implemented a set of changes in Vulnerability Detector that make it save the scan results for each agent in a new table within the agents' databases. This information allows a user to know which CVEs affect each package installed in the endpoint where the agent is installed.

Having said that, all the necessary changes should be implemented in the framework in order to make a user capable of querying this information by using the RESTful API.

The information in the new table will be defined as part of the issue https://github.com/wazuh/wazuh/issues/7512. Check the issue and comments for details.

In addition, the usage of this new capability should be documented in the RESTful API documentation.

### PR DESCRIPTION
**Pull request by Selutario**
|Related issue|
|---|
| Closes #7520 |

## Description

Hello team!

As requested in #7520, this PR adds a new endpoint that can be used to query the vulnerabilities of any agent. 

### Get first 3 items
```JSON
curl -k -X GET "https://localhost:55000/vulnerabilities/000?limit=3&pretty=true" -H  "Authorization: Bearer $TOKEN"

{
   "data": {
      "affected_items": [
         {
            "cve": "CVE-2020-27350",
            "version": "1.6.12ubuntu0.1",
            "architecture": "amd64",
            "name": "apt"
         },
         {
            "cve": "CVE-2020-35493",
            "version": "2.30-21ubuntu1~18.04.4",
            "architecture": "amd64",
            "name": "binutils"
         },
         {
            "cve": "CVE-2020-35495",
            "version": "2.30-21ubuntu1~18.04.4",
            "architecture": "amd64",
            "name": "binutils"
         }
      ],
      "total_affected_items": 39,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected vulnerabilities were returned",
   "error": 0
```

### Filter CVE where architecture = amd64
```JSON
curl -k -X GET "https://localhost:55000/vulnerabilities/000?limit=3&architecture="amd64"&pretty=true" -H  "Authorization: Bearer $TOKEN"
{
   "data": {
      "affected_items": [
         {
            "cve": "CVE-2020-27350",
            "version": "1.6.12ubuntu0.1",
            "architecture": "amd64",
            "name": "apt"
         },
         {
            "cve": "CVE-2020-35493",
            "version": "2.30-21ubuntu1~18.04.4",
            "architecture": "amd64",
            "name": "binutils"
         },
         {
            "cve": "CVE-2020-35495",
            "version": "2.30-21ubuntu1~18.04.4",
            "architecture": "amd64",
            "name": "binutils"
         }
      ],
      "total_affected_items": 38,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected vulnerabilities were returned",
   "error": 0
}
```

### Filter architecture = amd64 and version<0.8
```JSON
curl -G --data-urlencode "q=architecture=amd64;version<0.8" -k -X GET "https://localhost:55000/vulnerabilities/000?limit=3&pretty=true" -H  "Authorization: Bearer $TOKEN"
{
   "data": {
      "affected_items": [
         {
            "version": "0.23.9-2",
            "name": "libp11-kit0",
            "architecture": "amd64",
            "cve": "CVE-2020-29361"
         },
         {
            "version": "0.23.9-2",
            "name": "libp11-kit0",
            "architecture": "amd64",
            "cve": "CVE-2020-29362"
         },
         {
            "version": "0.23.9-2",
            "name": "libp11-kit0",
            "architecture": "amd64",
            "cve": "CVE-2020-29363"
         }
      ],
      "total_affected_items": 3,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected vulnerabilities were returned",
   "error": 0
}
```

Regards,
Selu.

### CODE REVIEW

File: api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from os import system, WEXITSTATUS
+import socket
+
+from base_healthcheck import get_manager_health_base
+
+if __name__ == "__main__":
+    # Workers are not needed in this test, so the exit code is set to 0 (healthy).
+    # os.system can return 256 (wait status), so WEXITSTATUS translate it to 'exit' status.
+    exit(WEXITSTATUS(system(
+        "grep -q 'wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '\\''002'\\''' /var/ossec/logs/ossec.log"))
```
Comment (davidjiglesias) [inline api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py:10]: agent 002?

File: api/test/integration/test_rbac_black_vulnerability_endpoints.tavern.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      status_code: 200
+      json:
+        error: 0
+        data:
+          affected_items: !anything
+          failed_items: []
+          total_affected_items: !anyint
+          total_failed_items: 0
+
+  - name: Try to get vulnerabilities information for manager (000)
```
Comment (davidjiglesias) [inline api/test/integration/test_rbac_black_vulnerability_endpoints.tavern.yaml:28]: No 000 pls

File: api/test/integration/test_rbac_white_all_endpoints.tavern.yaml [hunk 3]
```diff
@@ -141,6 +141,20 @@ stages:
     response:
       <<: *permission_denied
 
+---
+test_name: GET /vulnerability/{agent_id}cve
+
+stages:
+  - name: Try to get agent 000 vulnerabilities
```
Comment (davidjiglesias) [inline api/test/integration/test_rbac_white_all_endpoints.tavern.yaml:148]: Same

File: api/test/integration/test_rbac_white_vulnerability_endpoints.tavern.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      params:
+        limit: 1
+    response: &permission_denied
+      status_code: 403
+      json:
+        error: 4000
+
+  - name: Get vulnerabilities information for manager (000)
```
Comment (davidjiglesias) [inline api/test/integration/test_rbac_white_vulnerability_endpoints.tavern.yaml:24]: Same

File: api/test/integration/test_vulnerability_endpoints.tavern.yaml [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+test_name: GET /vulnerability/{agent_id}
+
+marks:
+  - base_tests
+
+stages:
+
+  # GET /vulnerabilities/000
+  - name: Get agent's vulnerabilities (limit and offset)
+    request: &cve_request_000
```
Comment (davidjiglesias) [inline api/test/integration/test_vulnerability_endpoints.tavern.yaml:11]: Same

File: framework/wazuh/core/vulnerability.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GP
+
+from wazuh.core import common
+from wazuh.core.agent import Agent
+from wazuh.core.utils import WazuhDBQuery, WazuhDBBackend
+
+
+class WazuhDBQueryCVE(WazuhDBQuery):
```
Comment (davidjiglesias) [inline framework/wazuh/core/vulnerability.py:10]: Prefer WazuhDBQueryVulnerability

File: framework/wazuh/rbac/default/policies.yaml [hunk 7]
```diff
@@ -330,3 +330,13 @@ default_policies:
         resources:
           - '*:*:*'
         effect: allow
+
+  vulnerability_read:
+    description: Allow read agents' vulnerabilities information.
```
Comment (davidjiglesias) [inline framework/wazuh/rbac/default/policies.yaml:335]:
```suggestion
    description: Allow reading agents' vulnerabilities information.
```

File: framework/wazuh/tests/data/security/rbac_catalog.yml [hunk 8]
```diff
@@ -799,6 +799,20 @@ get_rbac_actions:
           effect: deny
         related_endpoints:
           - GET /tasks/status
+      vulnerability:read:
+        description: Allow read agents' vulnerabilities information
```
Comment (davidjiglesias) [inline framework/wazuh/tests/data/security/rbac_catalog.yml:803]:
```suggestion
        description: Allow reading agents' vulnerabilities information
```

File: framework/wazuh/vulnerability.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    -------
+    result : AffectedItemsWazuhResult
+        JSON containing the vulnerabilities.
+    """
+    result = AffectedItemsWazuhResult(all_msg='All selected vulnerabilities were returned',
+                                      some_msg='Some vulnerabilities were not returned',
+                                      none_msg='No vulnerabilities were returned'
+                                      )
+
+    db_query = WazuhDBQueryCVE(agent_id=agent_list[0], offset=offset, limit=limit, sort=sort, search=search,
```
Comment (davidjiglesias) [inline framework/wazuh/vulnerability.py:50]: If agent_list is empty, this can return all, careful.

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
CVE files?
vulnerability ossec_conf always?

File: CHANGELOG.md [hunk 10]
```diff
@@ -15,6 +15,7 @@ All notable changes to this project will be documented in this file.
 
 - **API:**
   - Added new endpoint to get agent stats from different components. ([#7128](https://github.com/wazuh/wazuh/issues/7128))
+  - Added new endpoint to get CVEs that affect an agent. ([#7520](https://github.com/wazuh/wazuh/issues/7520))
```
[the remark this answers is not in the document]
Comment (Selutario) [reply CHANGELOG.md]: Changelog update was removed from this PR.

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**